pkg/controller/statefulset/stateful_set_utils.go

```diff
@@ -587,8 +587,8 @@ func inconsistentStatus(set *apps.StatefulSet, status *apps.StatefulSetStatus) b
 // are set to 0.
 func completeRollingUpdate(set *apps.StatefulSet, status *apps.StatefulSetStatus) {
 	if set.Spec.UpdateStrategy.Type == apps.RollingUpdateStatefulSetStrategyType &&
-		status.UpdatedReplicas == status.Replicas &&
-		status.ReadyReplicas == status.Replicas {
+		status.UpdatedReplicas == *set.Spec.Replicas &&
+		status.ReadyReplicas == *set.Spec.Replicas {
```

Comment:
Nit: I'm wondering if we might also need a check for `status.Replicas == *set.Spec.Replicas` and/or `status.CurrentReplicas == 0`? 

I don't _think_ we can get here now if we have condemned pods that haven't been successfully removed, but given the complexity of this controller's logic and the strong guarantee we need from this check (to avoid unexpected rollout progress), I think it might make it more resilient to check all conditions we care about.

Reply:
I agree, I will add this check too

Reply:
added that check kindly check